### ISSUE TITLE
PhysicsEnginePlatformer infinity loop

### ISSUE THREAD

**Issue by dzervas** [pvcraven/arcade#128]
```
                    while check_for_collision(self.player_sprite, item):
                        self.player_sprite.bottom = item.top
```
at line 133 in `arcade/physics_engines.py` infinity loops when my `AnimatedWalkingSprite` falls on a platform. The problem is that `item.top` is 32 and `self.player_sprite.bottom` is 31.99999999996 :)

Took ~8 hours to find that...

Thing is, I can't quite follow the math behind the engine. Can someone help me solve it?

**Comment by pvcraven** [comment]
I've spent a lot of time with rounding errors and that section of code. Looks like it needs more work.

**Comment by dzervas** [comment]
Ugh, I know...
Spent the whole weekend just to find where the hell stuff are not rounded...
Do you think I fixed it in #129? Or it's somewhere deeper we need to look?

**Comment by pvcraven** [comment]
Closed, implemented in 1.2.2 which is now deployed on PyPi.
